yt/frontends/gadget/io.py

```diff
@@ -242,7 +242,7 @@ def __init__(self, ds, *args, **kwargs):
         self._fields = ds._field_spec
         self._ptypes = ds._ptype_spec
         self.data_files = set([])
-        gformat = _get_gadget_format(ds.parameter_filename)
+        gformat = _get_gadget_format(ds.parameter_filename, ds._header_size)
```

Comment:
@MatthewTurk This is the bug I referred to in the commit message. It's an error made by myself that I forgot to pass the header size through.